torch/_numpy/_funcs_impl.py

```diff
@@ -360,11 +360,11 @@ def arange(
     # the dtype of the result
     if dtype is None:
         dtype = _dtypes_impl.default_dtypes().int_dtype
-    # XXX: default values do not get normalized
-    start, stop, step = (_util._coerce_to_tensor(x) for x in (start, stop, step))
-
-    dummy = torch.empty(1, dtype=dtype)
-    target_dtype = _dtypes_impl.result_type_impl(start, stop, step, dummy)
+        # XXX: default values do not get normalized
+        start, stop, step = (_util._coerce_to_tensor(x) for x in (start, stop, step))
```

Comment:
Can we avoid coercing these into tensors? I think that in this case we will not need to do anything and PyTorch will do the right thing just by passing in the scalars (modulo some `float32` vs `float64` correction.

Reply:
The issue is (of course) numpy scalars: `np.arange(np.int32(5))`.

The right thing to do is likely to have a special `ScalarOr0DArray`  type annotation and  a corresponding normalizer, throughout the codebase. Let's keep it on the backburner ATM.

Reply:
Already using ScalarOrArray here should work, as `torch.arange` will yell at you if the tensor you pass to it is not a 0D CPU tensor.